modules/prebidmanagerAnalyticsAdapter.js

```diff
@@ -0,0 +1,193 @@
+  track({eventType, args}) {
+    handleEvent(eventType, args);
+  }
+});
+
+prebidmanagerAnalytics.originEnableAnalytics = prebidmanagerAnalytics.enableAnalytics;
+prebidmanagerAnalytics.enableAnalytics = function (config) {
+  initOptions = config.options;
+  initOptions.url = initOptions.url || DEFAULT_EVENT_URL;
+  pmAnalyticsEnabled = true;
+  prebidmanagerAnalytics.originEnableAnalytics(config);
+};
+
+function flush() {
+  if (!pmAnalyticsEnabled) {
+    return;
+  }
+
+  if (_eventQueue.length > 1) {
+    var data = {
+      pageViewId: _pageViewId,
+      ver: _VERSION,
+      bundleId: initOptions.bundleId,
+      events: _eventQueue
+    };
+
+    ajax(
+      initOptions.url,
+      () => utils.logInfo(`${analyticsName} sent events batch`),
+      _VERSION + ':' + JSON.stringify(data),
+      {
+        contentType: 'text/plain',
+        method: 'POST',
+        withCredentials: true
+      }
+    );
+    _eventQueue = [
+      _pageView
+    ];
+  }
+}
+
+function handleEvent(eventType, eventArgs) {
+  eventArgs = eventArgs ? JSON.parse(JSON.stringify(eventArgs)) : {};
+  var pmEvent = {};
+
+  switch (eventType) {
+    case CONSTANTS.EVENTS.AUCTION_INIT: {
+      pmEvent = eventArgs;
+      _startAuction = pmEvent.timestamp;
+      _bidRequestTimeout = pmEvent.timeout;
+      break;
+    }
+    case CONSTANTS.EVENTS.AUCTION_END: {
+      pmEvent = eventArgs;
+      pmEvent.start = _startAuction;
+      pmEvent.end = Date.now();
+      break;
+    }
+    case CONSTANTS.EVENTS.BID_ADJUSTMENT: {
+      pmEvent.bidders = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.BID_TIMEOUT: {
+      pmEvent.bidders = eventArgs;
+      pmEvent.duration = _bidRequestTimeout;
+      break;
+    }
+    case CONSTANTS.EVENTS.BID_REQUESTED: {
+      pmEvent = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.BID_RESPONSE: {
+      pmEvent = eventArgs;
+      delete pmEvent.ad;
+      break;
+    }
+    case CONSTANTS.EVENTS.BID_WON: {
+      pmEvent = eventArgs;
+      delete pmEvent.ad;
+      delete pmEvent.adUrl;
+      break;
+    }
+    case CONSTANTS.EVENTS.BIDDER_DONE: {
+      pmEvent = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.SET_TARGETING: {
+      pmEvent.targetings = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.REQUEST_BIDS: {
+      pmEvent = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.ADD_AD_UNITS: {
+      pmEvent = eventArgs;
+      break;
+    }
+    case CONSTANTS.EVENTS.AD_RENDER_FAILED: {
+      pmEvent = eventArgs;
+      break;
+    }
+    default:
+      return;
+  }
+
+  pmEvent.eventType = eventType;
+  pmEvent.timestamp = pmEvent.timestamp || Date.now();
+
+  sendEvent(pmEvent);
+}
+
+function sendEvent(event) {
+  _eventQueue.push(event);
+  utils.logInfo(`${analyticsName}Event ${event.eventType}:`, event);
+
+  if (event.eventType === CONSTANTS.EVENTS.AUCTION_END) {
+    flush();
+  }
+}
+
+setInterval(flush, 1000);
```

Comment:
I think this can be moved inside `enableAnalytics` function. So it will only get called when this module is enabled

Reply:
Moved to `enableAnalytics `function